File: web/src/models/DTOs/Cobrancas/StatusInterface.ts
export interface StatusInterfaceDTO {
  descricao: string;
  nomeEnum: string;
}

export interface StatusInterfaceType {
  ABERTO: StatusInterfaceDTO;
  MEDICAO: StatusInterfaceDTO;
  COBRADO: StatusInterfaceDTO;
  AGRUPADO: StatusInterfaceDTO;
  PAGO: StatusInterfaceDTO;
  CANCELADO: StatusInterfaceDTO;
  CANCELAR: StatusInterfaceDTO;
  EXCLUIDO: StatusInterfaceDTO;
  EXTINGUIDO: StatusInterfaceDTO;
  EXTINGUIR: StatusInterfaceDTO;
  NAO_ADM_CR5: StatusInterfaceDTO;
  SUBSTITUIDO: StatusInterfaceDTO;
  SUBSTITUIR: StatusInterfaceDTO;
  FINANCIADO: StatusInterfaceDTO;
  NAO_EFETIVADO: StatusInterfaceDTO;
}

export type StatusInterface = keyof StatusInterfaceType;

/**
 * Aqui, label é um nome amigável e value é o valor no Enum StatusInterface no cr5-webservices-v2.
 * Constante feita para ser usada como opção de SelectControl
 */
export const formasDePagamentoValues: StatusInterfaceType = {
  ABERTO: {
    nomeEnum: "ABERTO",
    descricao: "Aberto",
  },
  MEDICAO: {
    nomeEnum: "MEDICAO",
    descricao: "Medicao",
  },
  COBRADO: {
    nomeEnum: "COBRADO",
    descricao: "Cobrado",
  },
  AGRUPADO: {
    nomeEnum: "AGRUPADO",
    descricao: "Agrupado",
  },
  PAGO: {
    nomeEnum: "PAGO",
    descricao: "Pago",
  },
  CANCELADO: {
    nomeEnum: "CANCELADO",
    descricao: "Cancelado",
  },
  CANCELAR: {
    nomeEnum: "CANCELAR",
    descricao: "Cancelar",
  },
  EXCLUIDO: {
    nomeEnum: "EXCLUIDO",
    descricao: "Excluido",
  },
  EXTINGUIDO: {
    nomeEnum: "EXTINGUIDO",
    descricao: "Extinguido",
  },
  EXTINGUIR: {
    nomeEnum: "EXTINGUIR",
    descricao: "Extinguir",
  },
  NAO_ADM_CR5: {
    nomeEnum: "NAO_ADM_CR5",
    descricao: "Não Administrado CR5",
  },
  SUBSTITUIDO: {
    nomeEnum: "SUBSTITUIDO",
    descricao: "Substituido",
  },
  SUBSTITUIR: {
    nomeEnum: "SUBSTITUIR",
    descricao: "Substituir",
  },
  FINANCIADO: {
    nomeEnum: "FINANCIADO",
    descricao: "Financiado",
  },
  NAO_EFETIVADO: {
    nomeEnum: "NAO_EFETIVADO",
    descricao: "Não Efetivado",
  },
};

File: webservices/src/main/java/fieg/modulos/interfacecobranca/enums/StatusInterface.java
package fieg.modulos.interfacecobranca.enums;

import fieg.core.interfaces.EnumBanco;
import fieg.modulos.boleto.enums.SituacaoBoleto;

public enum StatusInterface implements EnumBanco {

    ABERTO("ABERTO"),
    MEDICAO("MEDICAO"),
    COBRADO("COBRADO"),
    AGRUPADO("AGRUPADO"),
    PAGO("PAGO"),
    CANCELADO("CANCELADO"),
    NAO_EFETIVADO("NAO_EFETIVADO"),

    @Deprecated
    CANCELAR("CANCELAR"),
    EXCLUIDO("EXCLUIDO"),
    EXTINGUIDO("EXTINGUIDO"),

    @Deprecated
    EXTINGUIR("EXTINGUIR"),
    NAO_ADM_CR5("NAOADMCR5"),
    SUBSTITUIDO("SUBSTITUIDO"),

    @Deprecated
    SUBSTITUIR("SUBSTITUIR"),
    FINANCIADO("FINANCIADO");

    private final String valorBanco;

    StatusInterface(String valorBanco) {
        this.valorBanco = valorBanco;
    }

    public static StatusInterface fromString(String situacao) {
        for (StatusInterface s : StatusInterface.values()) {
            if (s.getValorBanco().equalsIgnoreCase(situacao)) {
                return s;
            }
        }
        throw new IllegalArgumentException("Situação inválida: " + situacao);
    }

    @Override
    public String getValorBanco() {
        return valorBanco;
    }
}
